modules/nextflow/src/test/groovy/nextflow/scm/AssetManagerTest.groovy

```diff
@@ -338,35 +337,22 @@ class AssetManagerTest extends Specification {
 
     }
 
-    def testCreateProviderFor(){
-
+    @Tag("core")
+    @Unroll
+    def 'should create hub provider for #providerName'() {
         when:
         def manager = new AssetManager()
-        def repo = manager.createHubProvider('github')
-        then:
-        repo instanceof GithubRepositoryProvider
-
-        when:
-        manager = new AssetManager()
-        repo = manager.createHubProvider('bitbucket')
-        then:
-        repo instanceof BitbucketRepositoryProvider
+        def repo = manager.createHubProvider(providerName)
 
-        when:
-        manager = new AssetManager()
-        repo = manager.createHubProvider('gitlab')
         then:
-        repo instanceof GitlabRepositoryProvider
-
-        when:
-        manager = [:] as AssetManager
-        manager.createHubProvider('xxx')
-        then:
-        thrown(AbortOperationException)
+        repo instanceof RepositoryProvider
 
+        where:
+        providerName << ['github', 'bitbucket', 'gitlab']
```

Comment:
It's not validating that for a given provider name is creating the expected provider class